From 993dd30854ecf0b8eacfbc2b005dfd59778da23d Mon Sep 17 00:00:00 2001
From: Petr Muller <muller@redhat.com>
Date: Wed, 10 Apr 2013 16:20:16 +0200
Subject: [PATCH 1/3] Do not use 'with open() as foo' construct

It does not work on RHEL<=5, with old Python not having
this syntax yet

fix for python compat on RHEL5
---
 src/python/bstor.py       | 5 +++--
 src/python/journalling.py | 5 +++--
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/python/bstor.py b/src/python/bstor.py
index 04d4dfd..a512795 100755
--- a/src/python/bstor.py
+++ b/src/python/bstor.py
@@ -36,8 +36,9 @@ class Storage(object):
 
   def __save_file(self, parser):
     fpath = os.path.join(self.__obtain_dir(), self.namespace)
-    with open(fpath, 'w') as cfile:
-      parser.write(cfile)
+    cfile = open(fpath, 'w')
+    parser.write(cfile)
+    cfile.close()
 
   def get(self, key):
     parser = self.__obtain_file()
diff --git a/src/python/journalling.py b/src/python/journalling.py
index 324a3ad..ed63e7f 100755
--- a/src/python/journalling.py
+++ b/src/python/journalling.py
@@ -440,8 +440,9 @@ class Journal(object):
 
     releaseEl   = newdoc.createElement("release")
     try:
-      with open("/etc/redhat-release", "r") as release_file:
-        release = release_file.read().strip()
+      release_file = open("/etc/redhat-release", "r")
+      release = release_file.read().strip()
+      release_file.close()
     except IOError:
       release = "unknown"
     release = unicode(release, 'utf-8', errors='replace')
-- 
1.9.3

